File: 1822. 数组元素积的符号/1822. 数组元素积的符号.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        def sign(n):
            if n==0:
                return 0
            elif n>0:
                return 1
            return -1
        res = 1
        for n in nums:
            res *= sign(n)
        return res

File: 1822. 数组元素积的符号/1822. 数组元素积的符号.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        // 统计负数的个数，根据个数返回符号
        int neg = 0;
        for (auto n:nums){
            if(n==0){
                return 0;
            }
            if (n<0){
                neg++;
            }
        }
        return neg%2 ? -1:1;
        
    }
};